### DISCUSSION TITLE
dbt Core

### DISCUSSION THREAD

**Discussion by amotl** [crate/crate-clients-tools#194] [Ideas]
## About

_Make adapter shippable, and reflect its capabilities._

## Details

I would like to introduce @hlcianfagna and @andnig to each other, because they seem to have a shared interest to make `dbt-cratedb` fly again. I think Hernan already has valuable improvements in some of his feature branches at [dbt-cratedb1.5](https://github.com/hlcianfagna/dbt-cratedb2). Hereby, we are aiming for a designated and canonical dbt-cratedb2.

- https://pypi.org/project/dbt-cratedb/
- https://github.com/crate-workbench/dbt-cratedb

## References
- https://github.com/crate/crate/issues/13234

**Comment by amotl** [comment]
## Status quo
The current canonical documentation is [Using dbt with CrateDB](https://community.cratedb.com/t/using-dbt-with-cratedb/1566) by @hlcianfagna and @hammerhead.

## Outlook
In an offline discussion, @hlcianfagna and me agreed to spend some time on making the dbt adapter shippable. After that, we may want to adjust the documentation accordingly.

**Comment by amotl** [comment]
Q: Can we do the same with CrateDB and its dbt adapter like what Snowflake is doing here? If so, my proposal is to also use »CrateDB Toolkit« to absorb what is presented here on behalf of »Snowpark ML Toolkit + dbt«.

> About what's Snowflake is doing, that is a nice summarizing article, also referencing a few previous ones of the same series. Coming from our initial articles on this topic, currently being published, do you think it would be possible to level up to be en par with those, @andnig and @ckurze?
>
> -- https://medium.com/snowflake/discover-the-new-snowpark-ml-toolkit-dbt-python-models-4f85b85a8c88
> -- https://github.com/Snowflake-Labs/sfguide-intro-to-machine-learning-with-snowpark-ml-for-python

> I did not investigate about their secret sauce yet, but I guess it is something [along the lines of Ibis](https://github.com/crate/roadmap/issues/23#issuecomment-1635600991). It also looks like they are also leveraging dbt, so working on something like that could be a team effort bundling and building upon a few of our previous individual endeavours around those topics.

-- via: https://crate.slack.com/archives/C05R7FU17EV/p1697221704129079

**Comment by hlcianfagna** [comment]
Hi,
https://github.com/crate-workbench/dbt-cratedb2/compare/main...hlcianfagna:dbt-cratedb2:hlcianfagna/2023MayUpdate
and 
https://github.com/crate-workbench/dbt-cratedb-examples/compare/main...hlcianfagna:dbt-cratedb-examples:hlcianfagna/2023MayUpdate
may be useful for this exercise.

Please note that since I last worked on that code CrateDB has added https://github.com/crate/crate/issues/14064 and https://github.com/crate/crate/pull/14953

**Comment by amotl** [comment]
Thanks. Do you suggest to ship your variant, no matter what the authors of https://pypi.org/project/dbt-cratedb/ are doing? Or do you think we should investigate once again on which details both variants show differences, maybe together?

I will be happy to follow both directions, depending on your suggestion.

**Comment by amotl** [comment]
Hi. After a conversation with @vvulf, I think we can count him in on the special interest group of people aiming to improve presentation of dbt technologies together with CrateDB.

**Comment by amotl** [comment]
Apparently, I already took the chance to translate the example repository [^1] by the author of [dbt-cratedb](https://pypi.org/project/dbt-cratedb/), [Julio Sánchez Jiménez](https://github.com/jsnchzjmnz), into English [^2] in preparation for a real "project". I just haven't used it yet, otherwise I would probably know more about dbt ;]. Maybe it could be used as a reasonable starting point, or even as the foundation for corresponding integration tests, when bringing all together?

Edit: I am only seeing now that @hlcianfagna already referenced it [^2] at https://github.com/crate/crate-clients-tools/issues/188, and that he also added a few commits over there.

[^1]: https://github.com/jsnchzjmnz/ejercicios-dbt-cratedb
[^2]: https://github.com/crate-workbench/dbt-cratedb-examples/

**Comment by amotl** [comment]
@geragray knows the CTO of dbt, and would like to get a corresponding well-maintained CrateDB adapter into the list of their trusted adapters, in order to improve visibility, co-marketing etc.

> I am talking to dbt labs CTO about getting into the trusted adapter pages. Going through criteria for trusted adapters, it seems like it is fairly easy to get in [^1]. It just needs to work plus maintained actively (by a company, not a community). 

[^1]: https://docs.getdbt.com/guides/adapter-creation?step=7

**Comment by amotl** [comment]
### Status Quo
dbt-cratedb2 exists now.

- **PyPI:** https://pypi.org/project/dbt-cratedb2/
- **GitHub:** https://github.com/crate/dbt-cratedb2
- **Documentation (onsite):** https://cratedb.com/docs/guide/integrate/dbt/
- **Documentation (remote):** https://docs.getdbt.com/docs/core/connect-data-platform/cratedb-setup

**Reply by amotl** [reply]
### Backlog 2024-12
- https://github.com/dbt-labs/docs.getdbt.com/pull/6564
- https://github.com/crate/dbt-cratedb2/issues/3
- https://github.com/crate/dbt-cratedb2/issues/4
